### DISCUSSION TITLE
Google Analytics 4 is replacing Universal Analytics

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2046] [Ideas]
**Today I received the message bellow from Google:**

"Google Analytics 4 is our next-generation measurement solution, and is replacing Universal Analytics. On July 1, 2023, Universal Analytics properties will stop processing new hits. If you still rely on Universal Analytics, we recommend that you complete your move to Google Analytics 4. 

Now's the time to create a new Google Analytics 4 property and ensure that your setup is complete. If you haven't started collecting data in your Google Analytics 4 property, you should do so now. This will allow you to gather the historical insights you need to measure your results over time when Universal Analytics stops processing hits. If you are already collecting the data you need in Google Analytics 4, you should ensure your remaining setup is complete.  
  
You can continue to use and collect new data in your Universal Analytics properties until July 1, 2023. After that, you will be able to access your previously processed data in Universal Analytics for a period of at least six months. We know your data is important to you and we strongly encourage you to export your historical reports during this time.  
  
In the coming months, we will provide a future date for when we will fully sunset Universal Analytics. On this sunset date, you will no longer be able to see your Universal Analytics reports in the Analytics interface or access your Universal Analytics data via the API.  
  
If you have questions on successfully making the switch to Google Analytics 4, check out our Help Center resources.  
  
See you online,  
The Google Analytics Team"

In OpenMage we use Google Analytics 3 (Universal Analytics). Do we need to do something about?

**Comment by fballiano** [comment]
I know I'm always a little bit like that 😅 but I'd directly remove the google analytics module, at the end of the day what it does is adding a few lines of javascript, anybody can do that in many other ways (for example I never used that module).

Fixing the issue in the proper was would mean create configurations to select analytics 3 or 4 and then generate the proper code ;-)

**Reply by addison74** [reply]
In OM we can select from the list two options (Universal Analytics and Google Analytics) which both work with the same account number. Unlike us who can manually add code to the page there are people who still use this easy option. For them it will be an issue when Google stops supporting the two variants. In Magento 2 they solved it recently.

No change is needed right now but on July 1, 2023 if this module remains in OM it will have to be restricted to a single variant. That list and parts of the source code need to be removed.

**Reply by tmotyl** [reply]
For ecommerce analitics should be more than just page views - conversion tracking, with cart/ chwckout page add to cart, reviews etc.
Not sure how much of it was covered in the old module

**Comment by addison74** [comment]
There is a big issue related to Google Analytics. In recent years more and more people are using extensions in browsers that block tracking scripts. I analyzed the raw log of the webserver for a week and identified IPs that looked like true visitors. Google considered only 1/3 true visits. Analyzing the logs with AWstats I got almost 13.000 visits (without bots) and Google counted 3100 visits. 

PS - I also use such extensions in the default browser to block tracking scripts. including GA.

**Comment by sreichel** [comment]
> In OpenMage we use Google Analytics 3 (Universal Analytics). **Do we need to do something about?**

Really good question.

On the one hand it would be nice to be up to date, on the other hand we (might) have to think about GDPR compliance - to do it properly.

**Comment by addison74** [comment]
A PR was created for this issue. More details here https://github.com/OpenMage/magento-lts/pull/3023.

**Comment by loekvangool** [comment]
Now that v20.1 is GA, congrats by the way. I'm still on v19 planning the upgrade but it's going to be complex. I was a bit surprised that the later GA4 events like `add_to_cart` were not backported to v19. Would that be a big effort not to do?

**Reply by fballiano** [reply]
No but the RFC was approved at that time and froze v19, you could still pretty easily port the googleanalytics module from v20 to v19 if you need to ;-)

**Reply by sreichel** [reply]
The RFC ... ruined the project. Smileys are  are inappropriate ...

**Comment by f1-outsourcing** [comment]
I was thinking of adding matomo stats for some basic statistics. Just because of thinking ga analytics is being blocked. This was more from personal experience, but never thought so many people are blocking this.

How/where should I add this javascript code of matomo?

**Reply by ma4nn** [reply]
I'm using my fork from here: https://github.com/vianetz/matomo-magento1

**Reply by hirale** [reply]
I created a module to send data from server-side to GA4.
https://github.com/hirale/openmage-ga4-measurement

**Reply by f1-outsourcing** [reply]
I have to rewrite this pluging/module. I am currently not using default identifiers like matomo and piwik, because these are blocked also by my ublock

**Reply by f1-outsourcing** [reply]
@ma4nn 
I think if you do not add this to the image tracking, matomo is not going to track anything.

`&rec=1` in script.phtml

**Reply by ma4nn** [reply]
@f1-outsourcing jep you're right, I was always using the version from `develop` that already had the proposed change. Just released a new version 1.2.0 of the extension with all the latest changes. Feel free to open an issue in case you encounter any other problems.
